### PR TITLE
weighted average of points in cornerdetection

### PR DESCRIPTION
**Pull request by Casper-NS**
Added a method for calculating a weighted average on a list of points. This is used to make the corners less jittery by using a weighted average of the last 20 points.

### CODE REVIEW

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package dk.scuffed.whiteboardapp.pipeline.stages.points_stages
+
+import dk.scuffed.whiteboardapp.pipeline.IPipeline
+import dk.scuffed.whiteboardapp.pipeline.stages.PointsOutputStage
+import dk.scuffed.whiteboardapp.utils.Vec2Float
+import dk.scuffed.whiteboardapp.utils.Vec2Int
+import kotlin.math.round
+
+internal class WeigthedPointsStage(
```
Comment (Mati-AAU) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt:9]: typo in weighted

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private val inputPoints: PointsOutputStage,
+    private val historySize: Int,
+    private val wieghtThreshold: Float,
```
Comment (Mati-AAU) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt:12]: typo in weighThreshold

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    pipeline: IPipeline)
+    : PointsOutputStage(pipeline, Vec2Int(0, 0), Vec2Int(0, 0), Vec2Int(0, 0), Vec2Int(0, 0))
+{
+    private var OldPoints = Array<Array<Vec2Int>>(4) {Array<Vec2Int>(historySize) { Vec2Int(0, 0) } }
```
Comment (S3JER) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt:16]:
```suggestion
    private var oldPoints = Array<Array<Vec2Int>>(4) {Array<Vec2Int>(historySize) { Vec2Int(0, 0) } }
```

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    private var pointIndex = 0
+
+
+    override fun update() {
+        OldPoints[0][pointIndex] = inputPoints.points[0]
+        OldPoints[1][pointIndex] = inputPoints.points[1]
+        OldPoints[2][pointIndex] = inputPoints.points[2]
+        OldPoints[3][pointIndex] = inputPoints.points[3]
```
Comment (S3JER) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt:24]:
```suggestion
        oldPoints[0][pointIndex] = inputPoints.points[0]
        oldPoints[1][pointIndex] = inputPoints.points[1]
        oldPoints[2][pointIndex] = inputPoints.points[2]
        oldPoints[3][pointIndex] = inputPoints.points[3]
```

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        var meanX = sumX / pointList.size
+        var meanY = sumY / pointList.size
+
+        var weightSum = 0.0
+        sumX = 0.0
+        sumY = 0.0
+
+        for (point in pointList){
```
Comment (S3JER) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt:56]: Maybe add some comments to tell what the for loop does

**Review by S3JER — CHANGES_REQUESTED** [review]

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt (hunk 5 above)
Comment (Casper-NS) [reply app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt]: fixed

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt (hunk 2 above)
Comment (Casper-NS) [reply app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt]: fixed

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt (hunk 1 above)
Comment (Casper-NS) [reply app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/points_stages/WeigthedPointsStage.kt]: fixed

**Review by S3JER — APPROVED** [review]
LGTM

**Pull request merged**